source/frontends/common2/CMakeLists.txt

```diff
@@ -17,37 +13,63 @@ set(HEADER_FILES
   commonframe.h
   commoncontext.h
   controllerdoublepress.h
-  gnuframe.h
-  fileregistry.h
-  ptreeregistry.h
-  programoptions.h
   utils.h
   timer.h
   speed.h
   )
 
+if (BUILD_APPLEN OR BUILD_QAPPLE OR BUILD_SA2)
+  set(SOURCE_FILES ${SOURCE_FILES}
+    fileregistry.cpp
+    gnuframe.cpp
+    programoptions.cpp
+    ptreeregistry.cpp
+  )
```

Comment:
these files are not needed for the libretro build, and allow boost to be conditionally linked